src/pcapi/scripts/autofill_business_unit_siret.py

```diff
@@ -0,0 +1,274 @@
+        if venue.siret:
+            business_unit_details["siret"].append(venue.siret)
+    return business_unit_details
+
+
+def have_bank_information(source):
+    return source.bankInformation and source.bankInformation.status == BankInformationStatus.ACCEPTED
+
+
+def get_business_unit_details_for_offerer(offerer):
+    business_unit_details = {}
+    offerer_bank_information_hash = None
+    if have_bank_information(offerer):
+        offerer_bank_information_hash = get_bank_information_hash(offerer.bankInformation)
+        business_unit_details[offerer_bank_information_hash] = build_business_unit_details(
+            offerer.bankInformation, offerer
+        )
+
+    offerer_venues = [venue for venue in offerer.managedVenues if not venue.businessUnitId]
+
+    for venue in offerer_venues:
+        if have_bank_information(venue):
+            bank_information_hash = get_bank_information_hash(venue.bankInformation)
+            if bank_information_hash in business_unit_details:
+                details = business_unit_details[bank_information_hash]
+                details["venues"].append(venue)
+            else:
+                details = build_business_unit_details(venue.bankInformation, offerer, venue)
+        elif offerer_bank_information_hash in business_unit_details:
+            details = business_unit_details[offerer_bank_information_hash]
+            details["venues"].append(venue)
+        else:
+            details = None
+
+        if details:
+            if venue.siret:
+                details["siret"].append(venue.siret)
+                details["siret"] = list(set(details["siret"]))
+            business_unit_details[get_bank_information_hash(details)] = details
+    return business_unit_details
+
+
+def get_main_venue_siren_api_data(offerer):
+    print("get_main_venue_siren_api_data offerer", offerer)
+    siren_api_data_for_offerer = get_by_offerer(offerer)
+    siren_api_main_venue_data = siren_api_data_for_offerer["unite_legale"]["etablissement_siege"]
+
+    return {
+        "name": siren_api_main_venue_data["enseigne_1"],
+        "address": siren_api_main_venue_data["geo_l4"],
+        "postalCode": siren_api_main_venue_data["code_postal"],
+        "city": siren_api_main_venue_data["libelle_commune"],
+        "latitude": siren_api_main_venue_data["latitude"],
+        "longitude": siren_api_main_venue_data["longitude"],
+        "siret": siren_api_main_venue_data["siret"],
+    }
+
+
+def get_offerer_main_venue_from_siren_api_data(offerer, siren_api_data):
+
+    main_venue = Venue.query.filter(
+        Venue.managingOffererId == offerer.id,
+        Venue.siret == siren_api_data["siret"],
+    ).one_or_none()
+
+    # try to find venue from address
+    if not main_venue:
+        main_venue = Venue.query.filter(
+            Venue.managingOffererId == offerer.id,
+            Venue.postalCode == siren_api_data["postalCode"],
+            Venue.city == siren_api_data["city"],
+        ).one_or_none()
```

Comment:
Je pense qu'il y a des offreurs qui ont plusieurs lieux dans la même ville. Le `one_or_none()` ici lèvera une erreur `MultipleResultsFound`. Suggestion : ne prendre un lieu que s'il n'y en a qu'un dans la ville.

```suggestion
    in_same_city = [venue for venue in offerer.managedVenues if venue.postalCode == siren_api_data["postalCode"] and venue.city == siren_api_data["city"]]
    if len(in_same_city) == 1:
        return in_same_city[0]
    return None
```

Reply:
On ne cherche pas sur la ville mais sur l'adresse exact car on veux l'unique lieu qui peu correspondre à adresse du siège sociale de la structure.

Reply:
Hmmm, mais ta requête SQL ne filtre que sur le code postal et la ville, **sans la rue**. Ou bien j'ai loupé quelque chose ?

Reply:
c'est une erreur de ma part, bientot j'aurai des count pour voir combien on a de match sur ces addresses